client/src/app/site/base/base-user-import.service.ts

```diff
@@ -1,13 +1,15 @@
 import { Directive } from '@angular/core';
 
-import { User } from '../../domain/models/users/user';
+import { GENDERS, User } from '../../domain/models/users/user';
 import { ImportServiceCollectorService } from '../services/import-service-collector.service';
 import { BaseImportService } from './base-import.service';
 
 @Directive()
 export abstract class BaseUserImportService extends BaseImportService<User> {
     public override requiredHeaderLength = 3;
 
+    private readonly currentLangGenders = GENDERS.map(gender => this.translate.instant(gender));
```

Comment:
This looks risky: depending on the loading order, the translations might not be loaded yet when this class is initialized. `translate.instant` does not take this into account and just returns the english word if no translations are loaded yet (see https://github.com/ngx-translate/core#methods). Maybe you should use `get`/`stream` instead and update it on updates to the translation.

Another thing: What happens if you change the language while on this page? That would probably break the import as well.

Reply:
Good point, I'll change it

Reply:
Although: About the "Change the language"-thing. As long as you have uploaded a file, that would probably happen anyway unless the uploaded file is in English, so I probably won't be going any further than changing the way currentLangGenders works.